rust/src/lockfile.rs

```diff
@@ -416,7 +416,12 @@ pub(crate) fn lockfile_write(
     }
 
     let filename = Path::new(filename);
-    let lockfile_dir = openat::Dir::open(filename.parent().unwrap_or_else(|| Path::new("/")))?;
+    let lockfile_dir = match filename.parent() {
+        Some(p) if matches!(p.to_str(), Some("")) => Path::new("."),
```

Comment:
Yes, this trap has bit me before too!  But if you don't mind...we're in the middle of a slow port to [cap-std](https://docs.rs/cap-std/latest/cap_std/) and specifically our little helper crate cap-std-ext has a [helper method for exactly this](https://docs.rs/cap-std-ext/latest/cap_std_ext/dirext/trait.CapStdExtDirExt.html#tymethod.replace_file_with_perms) that gets this right.

Mind doing that port here instead?  You can find examples of other uses of this API in our code too.

Reply:
For reference, equivalent logic: https://github.com/coreos/cap-std-ext/blob/f07cacc899e0faf0f6ba105050baf416fd8811c7/src/tempfile.rs#L73

Reply:
All done, let me know what you think.

Reply:
It can be a lot simpler than that; I pushed an update commit - does that look OK to you?  I tested it.

Perhaps we should add a helper to get a `Dir` from a filename though.